### PR TITLE
Add/update site configs for s4, discover, cheyenne, default

### PR DESCRIPTION
**Pull request by climbfuji**
### Description

This PR adds/updates the site configs for University of Wisconsin S4, NASA Discover, NCAR-Wyoming Cheyenne.

### Testing

With these changes, spack-stack builds on Cheyenne, S4 and Discover.

CI tests currently running at https://github.com/climbfuji/spack-stack/actions.

### Dependent PRs

- waiting on https://github.com/NOAA-EMC/spack/pull/36

### CODE REVIEW

File: .github/actions/setup-spack-stack/action.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ inputs:
   mpi:
     description: 'Which MPI flavor (openmpi, mpich, intel-oneapi-mpi)'
     required: false
-    default: 'mpich'
+    default: 'mpich@3'
```
Comment (climbfuji) [inline .github/actions/setup-spack-stack/action.yaml]: Note to self. This can be reverted.

File: .github/actions/setup-spack-stack/action.yaml [hunk 2]
```diff
@@ -84,17 +84,21 @@ runs:
       source setup.sh
       spack env activate ${{ inputs.path }}/envs/${{ inputs.app }}
 
-      spack config add "packages:all:compiler:[${{ inputs.compiler }}]"
-      spack config add "packages:all:providers:mpi:[${{ inputs.mpi }}]"
+      spack config add "packages:all:compiler::[${{ inputs.compiler }}]"
```
Comment (climbfuji) [inline .github/actions/setup-spack-stack/action.yaml]: Note to self. Double colons don't work in `spack.yaml`, revert.

File: .github/actions/setup-spack-stack/action.yaml [hunk 3]
```diff
@@ -84,17 +84,21 @@ runs:
...
+      spack config add "packages:all:providers:mpi::[${{ inputs.mpi }}]"
 
       cd ${{ inputs.path }}/envs/${{ inputs.app }}
 
+      # Whitelist the mpi providers so that spack creates the modules for them
```
Comment (climbfuji) [inline .github/actions/setup-spack-stack/action.yaml:101]: Note to self. Double colons don't work in spack.yaml, change.

File: .github/actions/setup-spack-stack/action.yaml [hunk 4]
```diff
@@ -84,17 +84,21 @@ runs:
...
+      spack config add "modules:default:tcl:whitelist::[${{ inputs.mpi }}]"
+      spack config add "modules:default:lmod:whitelist::[${{ inputs.mpi }}]"
+
       if [ "$RUNNER_OS" == "Linux" ]; then
         echo ""
       elif [ "$RUNNER_OS" == "macOS" ]; then
-        spack config add "packages:boost:version:[1.78.0]"
-        spack config add "packages:fms:variants: +64bit +enable_quad_precision +gfs_phys ~openmp +pic"
-        spack config add "packages:fms-jcsda:variants: +64bit +enable_quad_precision +gfs_phys ~openmp +pic"
+        spack config add "packages:boost:version::[1.78.0]"
```
Comment (climbfuji) [inline .github/actions/setup-spack-stack/action.yaml]: Note to self. Double colons don't work in spack.yaml, revert.

File: .github/actions/setup-spack-stack/action.yaml [hunk 5]
```diff
@@ -126,3 +131,8 @@ runs:
       source setup.sh
       spack env activate ${{ inputs.path }}/envs/${{ inputs.app }}
       meta_modules/setup_meta_modules.py
+      # Test loading the meta modules
+      module use $PWD/envs/${{ inputs.app }}/install/modulefiles/Core
+      module load stack-${{ inputs.compiler }}
+      module load stack-python
+      module load stack-${{ inputs.mpi }}
```
Comment (climbfuji) [inline .github/actions/setup-spack-stack/action.yaml]: Fix.

File: .github/workflows/macos-gcc.yaml [hunk 6]
```diff
@@ -10,7 +10,8 @@ jobs:
   build:
     strategy:
       matrix:
-        app: [jedi-ufs, jedi-um, jedi-fv3, jedi-tools, ufs-weather-model, nceplibs]
+        #app: [jedi-ufs, jedi-um, jedi-fv3, jedi-tools, ufs-weather-model, nceplibs]
+        app: [test]
```
Comment (climbfuji) [inline .github/workflows/macos-gcc.yaml]: Revert.

File: configs/repos/jedi/packages/fckit/libstdc++_new.patch [hunk 7]
```diff
@@ -0,0 +1,92 @@
+--- a/src/tests/CMakeLists.txt	2022-03-22 16:56:28.000000000 -0600
```
Comment (climbfuji) [inline configs/repos/jedi/packages/fckit/libstdc++_new.patch]: Not needed - remove and revert `package.py`.

File: configs/sites/discover/README.md [hunk 8]
```diff
@@ -1,5 +1,24 @@
...
+# Discover
+
+## General instructions/prerequisites
+
+module purge
+
+### Set up the user environment for working with spack/building new software environments
+This needs to be done every time before installing packages with spack or before using spack-provided modules!
+NO ONLY FOR BUILDING?
+
```
Comment (climbfuji) [inline configs/sites/discover/README.md]: WIP

File: configs/sites/discover/compilers.yaml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    flags: {}
+    operating_system: sles12
+    target: x86_64
+    modules:
+    - comp/intel/2021.5.0
+    environment:
+      prepend_path:
+        PATH: '/usr/local/other/gcc/10.1.0/bin'
+        LD_LIBRARY_PATH: '/usr/local/other/gcc/10.1.0/lib64'
+        #CPATH: '/usr/local/other/gcc/10.1.0/include:/usr/local/intel/oneapi/2021/mpi/2021.5.0/include'
```
Comment (climbfuji) [inline configs/sites/discover/compilers.yaml]: Need line 18?

**Comment by climbfuji** [comment]
CI tests are running here: https://github.com/climbfuji/spack-stack/actions

**Comment by climbfuji** [comment]
@kgerheiser This is ready to merge, finally. There are also two or so spack PRs that are ready and can be merged independently of this one. CI tests for this PR passed except the last metamodule step for Intel (https://github.com/climbfuji/spack-stack/actions).

**Review by kgerheiser — APPROVED** [review]

**Comment by kgerheiser** [comment]
Could you update the submodule with the most recent NCO (and other) PR?

**Comment by climbfuji** [comment]
> Could you update the submodule with the most recent NCO (and other) PR?

Done. The CI tests I linked above are still good in my opinion, thus I'll merge.

**Pull request merged**